graph/pull_v2.go

```diff
@@ -74,14 +74,22 @@ func (p *v2Puller) pullV2Repository(tag string) (err error) {
 	}
 
 	broadcaster, found := p.poolAdd("pull", taggedName)
+	broadcaster.Add(p.config.OutStream)
 	if found {
 		// Another pull of the same repository is already taking place; just wait for it to finish
-		broadcaster.Add(p.config.OutStream)
-		broadcaster.Wait()
+		result := broadcaster.Wait()
+		if err, ok := result.(error); ok {
+			return err
+		}
 		return nil
 	}
-	defer p.poolRemove("pull", taggedName)
-	broadcaster.Add(p.config.OutStream)
+	defer func() {
+		if err != nil {
```

Comment:
Actually is it, saw it expects `interface{}`, if you pass in a nil error vs nil interface{} is it treated the same? If this is required should add a comment.